src/main/java/com/game/Battleship/controllers/BoardController.java

```diff
@@ -1,17 +1,52 @@
 package com.game.Battleship.controllers;
-
 import com.game.Battleship.models.Board;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RestController;
+import com.game.Battleship.models.Ship;
+import com.game.Battleship.repository.BoardRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+import java.util.List;
+import java.util.Map;
 
 @RestController
 public class BoardController {
 
-    @GetMapping("/board")
-    public Board getBoard(){
+    @Autowired
+    private BoardRepository boardRepository;
+
+    @GetMapping("/boards")
+    public List<Board> getAllBoards(){
+
+        return boardRepository.allBoards();
+
+    }
+
+    //json object example: {"row": 0, "column": 1, "shipType": "Cruiser", "shipSize": 3, "isHorizontal": false }
+    @PatchMapping(value = "/boards/place")
+    public Board placeShip(@RequestBody Map<String, Object> changes) {
+
+        Board board = boardRepository.getPlayerBoard();
+
+        Integer row = ( Integer ) changes.get("row");
+        Integer column = ( Integer ) changes.get("column");
+        String shipType = ( String) changes.get("shipType");
+        Integer shipSize = (Integer) changes.get("shipSize");
+        Boolean isHorizontal = ( Boolean ) changes.get("isHorizontal");
+        Ship ship = new Ship(shipType, shipSize);
+        ship.setHorizontal(isHorizontal);
+
+        return boardRepository.updateBoardWithShips(board, ship, row, column);
```

Comment:
Doesn't the `BoardRepository` already have knowledge of `PlayerBoard`?

Reply:
yep..changed it!